app/decorators/field_service_instance_organization_decorator.rb

```diff
@@ -12,9 +12,9 @@ def initialize(fields)
       @fields = fields[:'space.organization'].to_set.intersection(self.class.allowed)
     end
 
-    def decorate(hash, service_instances)
+    def decorate(hash, resources)
       hash[:included] ||= {}
-      spaces = service_instances.map(&:space).uniq
+      spaces = resources.map { |r| r.try(:space) || r }.uniq
```

Comment:
I am not a fan of this in either of the two decorators. It looks to me more like accidental duplication and would have probably written them again as we might want to include different fields for space vs shared space/org vs shared org in the future. I am ok keeping this anyway, but would split them out as soon as they start deviating from each other.

Reply:
Yeah, it's a bit weird, I agree. We can probably write something better that does not involve duplicating this code entirely.

I don't see the divergence of fields as a problem though. The message we create in the controller should determine what each method can include, returning earlier if the request includes non-allowed fields (that is, if the message is invalid).

You can see that the `relationships_shared_spaces`, `index` and `show` all use this same decorator, but not necessarily allow the same fields to be included.

Reply:
hmm, you are right. I always forget it is restricted in the message anyway. And also agree we can write something better than duplicating the whole code as well. The only thing I can think of at the moment is inheritance and having a get_spaces method overriden by subclasses. It feels overkilling though.

Reply:
I'll merge this for now and we can always get back 👀